ignite/distributed/utils.py

```diff
@@ -737,3 +740,11 @@ def download_dataset():
 
     if current_rank == rank:
         barrier()
+
+
+def _rank_not_in_group(group: Optional[dist.ProcessGroup | "hvd.ProcessSet"]) -> bool:
```

Comment:
```suggestion
def _rank_not_in_group(group: Optional[Union[Any, List[int]]]) -> bool:
```
maybe this type hint instead?